scripts/pre_commit_linter.py

```diff
@@ -0,0 +1,153 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Pre-commit script for oppia
+This script checks Javascript coding style using jscs node module
+It returns and fix errors
+
+IMPORTANT NOTES:
+
+1.  Before running this script, you must install third-party dependencies by
+    running
+
+        bash scripts/start.sh
+
+    at least once.
+
+2.  This script should be run from the oppia root folder:
+
+        python scripts/pre_commit_linter.py
+ Note that the root folder MUST be named 'oppia'.
+ """
+
+__author__ = "Barnabas Makonda(barnabasmakonda@gmail.com)"
+
+import os
+import subprocess
+import sys
+
+
+def _get_list_of_current_changed():
+    """This function combines all changed files staged
+    and not staged
+    Returns:
+        a list of files about to be commited. """
+    changed_and_staged_files = subprocess.check_output(['git', 'diff', '--name-only'])
```

Comment:
Actually I take that back -- this is just a big string, right? In that case it may be clearer to do

    unstaged_filenames = subprocess.check_output([
        'git', 'diff', '--name-only']).splitlines()

and similarly for staged_filenames.